src/claimtrie.cpp

```diff
@@ -963,164 +625,41 @@ bool CClaimTrie::updateTakeoverHeight(const std::string& name, int nTakeoverHeig
-        }
-    }
-}
-
-void CClaimTrie::BatchWriteQueueNameRows(CDBBatch& batch)
-{
-    for (queueNameType::iterator itQueue = dirtyQueueNameRows.begin(); itQueue != dirtyQueueNameRows.end(); ++itQueue)
-    {
-        if (itQueue->second.empty())
-        {
-            batch.Erase(std::make_pair(CLAIM_QUEUE_NAME_ROW, itQueue->first));
-        }
-        else
-        {
-            batch.Write(std::make_pair(CLAIM_QUEUE_NAME_ROW, itQueue->first), itQueue->second);
-        }
-    }
-}
-
-void CClaimTrie::BatchWriteExpirationQueueRows(CDBBatch& batch)
-{
-    for (expirationQueueType::iterator itQueue = dirtyExpirationQueueRows.begin(); itQueue != dirtyExpirationQueueRows.end(); ++itQueue)
-    {
-        if (itQueue->second.empty())
-        {
-            batch.Erase(std::make_pair(EXP_QUEUE_ROW, itQueue->first));
-        }
-        else
-        {
-            batch.Write(std::make_pair(EXP_QUEUE_ROW, itQueue->first), itQueue->second);
-        }
-    }
-}
-
-void CClaimTrie::BatchWriteSupportNodes(CDBBatch& batch)
-{
-    for (supportMapType::iterator itSupport = dirtySupportNodes.begin(); itSupport != dirtySupportNodes.end(); ++itSupport)
-    {
-        if (itSupport->second.empty())
-        {
-            batch.Erase(std::make_pair(SUPPORT, itSupport->first));
-        }
-        else
-        {
-            batch.Write(std::make_pair(SUPPORT, itSupport->first), itSupport->second);
-        }
-    }
-}
-
-void CClaimTrie::BatchWriteSupportQueueRows(CDBBatch& batch)
-{
-    for (supportQueueType::iterator itQueue = dirtySupportQueueRows.begin(); itQueue != dirtySupportQueueRows.end(); ++itQueue)
-    {
-        if (itQueue->second.empty())
-        {
-            batch.Erase(std::make_pair(SUPPORT_QUEUE_ROW, itQueue->first));
-        }
-        else
-        {
-            batch.Write(std::make_pair(SUPPORT_QUEUE_ROW, itQueue->first), itQueue->second);
-        }
-    }
-}
-
-void CClaimTrie::BatchWriteSupportQueueNameRows(CDBBatch& batch)
+bool CClaimTrie::WriteToDisk()
 {
-    for (queueNameType::iterator itQueue = dirtySupportQueueNameRows.begin(); itQueue != dirtySupportQueueNameRows.end(); ++itQueue)
+    for (nodeCacheType::iterator itcache = dirtyNodes.begin(); itcache != dirtyNodes.end(); ++itcache)
     {
-        if (itQueue->second.empty())
-        {
-            batch.Erase(std::make_pair(SUPPORT_QUEUE_NAME_ROW, itQueue->first));
-        }
-        else
-        {
-            batch.Write(std::make_pair(SUPPORT_QUEUE_NAME_ROW, itQueue->first), itQueue->second);
+        CClaimTrieNode *pNode = itcache->second;
+        uint32_t num_claims = pNode ? pNode->claims.size() : 0;
+        LogPrintf("%s: Writing %s to disk with %d claims\n", __func__, itcache->first, num_claims);
+        if (pNode) {
+            db.updateQueueRow(itcache->first, *pNode);
+        } else {
+            CClaimTrieNode emptyNode;
+            db.updateQueueRow(itcache->first, emptyNode);
```

Comment:
Alternative style:
```
CClaimTrieNode emptyNode;
db.updateQueueRow(itcache->first, pNode ? *pNode : emptyNode);
```

I'll stop commenting on style, perhaps it's pointless, or maybe clang-format will help (although not on things like this, which are adding unnecessary lines).